src/osd/PG.cc

```diff
@@ -4168,7 +4168,9 @@ void PG::chunky_scrub(ThreadPool::TPHandle &handle)
 	  done = true;
 	  break;
 	}
-
+	if (state_test(PG_STATE_REPAIR)) {
+	  scrub_process_inconsistent();
```

Comment:
As I see it, there are two sections of scrub logic: a per-chunk part (scrub_compare_maps) and a per-scrub part (scrub_finish).  What I see here is that you are trying to move scrub_process_inconsistent() up to the per-chunk part, but only for REPAIR.  That seems brittle.  I'd rather you remove scrub_process_inconsistent from scrub_finish and assert in scrub_finish that scrubber.authoritative, etc, are cleared.  Is there some reason to leave it at the end for non-repair scrubs?